### DISCUSSION TITLE
Big latency during filtering

### DISCUSSION THREAD

**Discussion by TyrionFront** [directus/directus#16305] [Archive]
During attempts to filter items of any big collection using filtering based on quantity of entity's related items of another entity, system just gets stuck. The conditions and appropriate result are on the screens:
- conditions
<img width="890" alt="Screenshot 2022-11-03 at 15 49 50" src="https://user-images.githubusercontent.com/18406903/199742659-6fdebe95-49e5-4fe7-af13-2c754673cabc.png">

-result - more than 6 and  a half mins.
<img width="891" alt="Screenshot 2022-11-03 at 15 57 15" src="https://user-images.githubusercontent.com/18406903/199742666-1a860f87-036d-4dee-82e7-9baefebbf081.png">

### To Reproduce

Just apply filtering on a big (1000+) collection as it is described on the screens above.

### What version of Node.js are you using?
v16.15.0

### What database are you using?
PostgreSQL 14.5 on aarch64-apple-darwin20.6.0

**Comment by rijkvanzanten** [comment]
This is very dependent on the exact server performance of the database. The SQL it runs under the hood is relatively heavy, as it's counting on a nested o2m set and then applying the filter. Counting in minutes is very rough though. Would you be able to share a database dump of this setup, or some other way to reproduce this so we can take a peak if there's other bottlenecks?

**Reply by TyrionFront** [reply]
With all the respect - sharing data is not an option.
But is there a way to see the raw query that is being built by Directus during this operation ?

**Reply by TyrionFront** [reply]
@rijkvanzanten  ?

**Reply by TyrionFront** [reply]
@rijkvanzanten I've sent you some additional info to [rijkvanzanten@me.com](mailto:rijkvanzanten@me.com). Would be appreciated for any help.

**Reply by rijkvanzanten** [reply]
I can't offer direct live support to self-hosted projects, so we'll take a look when we can 👍🏻 We haven't seen this level of perf degradation in our own cloud offering testing, so there's some combination of infrastructure configuration and exact queries executed. You can see the sql queries that are executed by setting LOG_LEVEL to `trace` 👍🏻

**Reply by TyrionFront** [reply]
Extremely appreciated for the hint!)
I understand all the moments related to different lvls of projects - NP at all.

**Reply by TyrionFront** [reply]
@rijkvanzanten it seems like the core reason of the delay is in subsequent selects with counting for every id from the collection that is being filtered (slugs). In my case it means that the final quantity of applied queries is **_43623_**. But what is really confusing is that for some reason there are 2 absolutely identical queries followed one by one (highlighted with a red rectangle).
<img width="872" alt="Screenshot 2022-11-09 at 10 22 00" src="https://user-images.githubusercontent.com/18406903/200813807-26dcfeeb-71e9-4ca9-90f2-f342094f6340.png">
<img width="961" alt="Screenshot 2022-11-09 at 10 28 02" src="https://user-images.githubusercontent.com/18406903/200813811-a128e987-b2f7-4b16-a888-a9e2fbc3c180.png">
